rust-libretro/src/environment.rs

```diff
@@ -788,34 +841,32 @@ pub unsafe fn set_game_geometry(
 /// If this environment callback is used by a core that requires a valid username,
 /// a default username should be specified by the core.
 #[proc::context(GenericContext)]
-pub unsafe fn get_username<'a>(callback: retro_environment_t) -> Option<&'a str> {
```

Comment:
`NULL` is a valid return value for `RETRO_ENVIRONMENT_GET_USERNAME`, so this should return something like a `Result<Option<&'a str>, EnvironmentCallError>`.